hal/src/sercom/v2/i2c/pads.rs

```diff
@@ -0,0 +1,128 @@
+//! Define a container for a set of SERCOM pads
+//!
+//! See the [i2c module](super) documentation for more details on declaring and
+//! instantiating a [`Pads`] type.
+
+use crate::sercom::v2::*;
+use crate::typelevel::{NoneT, Sealed};
+use core::marker::PhantomData;
+
+pub struct Pads<S, SDA, SCL>
+where
+    S: Sercom,
+    SDA: IsPad<PadNum = Pad0, Sercom = S> + IsI2cPad,
+    SCL: IsPad<PadNum = Pad1, Sercom = S> + IsI2cPad,
```

Comment:
Actually, I think there might be some room for improvement of the existing structure. I think both `IsPad` and `GetPad` could have the `IoSet` as an associated type. I might not have realized that before. I'll play with it. That would clean up a few trait bounds.

Reply:
Had another realization just now. For I2C on `thumbv7em,` `Sercom` and `IoSet` is enough information to specify the `Pads`. That's not true for SPI, though, because you can mix and match MISO and MOSI to different `PadNum`s.

I think you'll need a different implementation anyway, so maybe it's worth trying to implement that. I think it would require a new trait though, to look up `PinId` by `Sercom` and `IoSet`.

Reply:
Ah, now I remember. `IoSet` can't be an associated type of `IsPad`, because the same `PinId` and `PinMode` can map to *multiple* `IoSet`s for the same `Sercom`.

Reply:
So I think it would be possible to implement `i2s::Pads<S, I> where S: Sercom, I: IoSet`. Knowing the `Sercom` and `IoSet` is definitely enough information, because you know to use `Pad0` and `Pad1`. *However*, I think we may not want to actually do that.

Number one, it would be an outlier among the different `Pads` structs, so it could add confusion. But secondly, I think it might make type inference worse. Right now, the compiler is able to infer **all** of the types when creating `uart::Pads` or `spi::Pads` directly in `main`. It can derive the `Sercom` type from the PAC `SERCOM` type. It can get the `PinId`s from the arguments to `.mosi()`, etc. It then looks up the correct `PinMode`. And finally, it can compare all of those to discover that only **one** `IoSet` type is common across all the pins.

If we switch to `Pads<S, I>`, the user **must** specify the `IoSet`, even in `main`, because it can never be inferred. At least it can't without analyzing multiple `PinId` types simultaneously.

It would also add additional type-level complexity to something that is already fairly complex. I think it's probably not worth it. What do you think?

Reply:
I'd rather use the approach we used in other sercom modules, I agree that it's not worth it. The regular approach seems to work quite well from the BSP examples I've included!